core/src/main/scala/org/apache/spark/deploy/SparkSubmitArguments.scala

```diff
@@ -43,7 +43,8 @@ private[deploy] class SparkSubmitArguments(args: Seq[String], env: Map[String, S
   extends SparkSubmitArgumentsParser with Logging {
   var maybeMaster: Option[String] = None
   // Global defaults. These should be keep to minimum to avoid confusing behavior.
-  def master: String = maybeMaster.getOrElse("local[*]")
+  def master: String =
+    maybeMaster.getOrElse(System.getProperty("spark.test.master", "local[*]"))
```

Comment:
The change is fine but just curious. Why can't we just set `spark.master` instead? The Python documentation build actually works fine in my mac fwiw.

Reply:
Yes, we can change it to accept `spark.master` here.
> Why can't we just set spark.master instead? 

For the following questions, I guess you already have https://github.com/apache/spark/pull/48129 to limit the number of SparkSubmit. It mitigate the situation. This PR and #48129 aims to two layers of Python documentations.
> The Python documentation build actually works fine in my mac fwiw.

Reply:
BTW, the reason why I didn't use `spark.master` directly, it could have some side effects in K8s side. Let me check that part to make ti sure, @HyukjinKwon .